rpmio/rpmpgp.c

```diff
@@ -330,6 +330,34 @@ struct pgpPkt {
     size_t blen;		/* length of body in bytes */
 };
 
+/** \ingroup rpmpgp
+ * Read a length field `nbytes` long.  Checks that the buffer is big enough to
+ * hold `nbytes + *valp` bytes.
+ * @param s		pointer to read from
+ * @param nbytes	length of length field
+ * @param send		pointer past end of buffer
+ * @param[out] *valp	decoded length
+ * @return		0 if buffer can hold `nbytes + *valp` of data,
+ * 			otherwise -1.
+ */
+static int pgpGet(const uint8_t *s, size_t nbytes, const uint8_t *send,
+		  size_t *valp)
+{
+    int rc = -1;
+
+    if (nbytes <= send - s &&
+	nbytes <= sizeof(size_t) &&
+	nbytes <= sizeof(unsigned int)) {
```

Comment:
This looks highly bogus, what does size of size_t have to do with anything PGP?
Okay, I guess the bogosity comes from pgpGrab() dealing with sizeof(size_t) which looks equally bogus to begin with, going back all the way to 95c55a1af9fa4be56737eefa0e709ce5abe47deb. Looks like well-intended but misguided attempt to size_t type-pedantry, I'm kinda suprised this isn't my fault because I certainly did my share of those in the early days :sweat_smile: 

Also the condition aligns on the same indentation level with the following code making it unnecessary hard to read.

Reply:
> This looks highly bogus, what does size of size_t have to do with anything PGP?
> Okay, I guess the bogosity comes from pgpGrab() dealing with sizeof(size_t) which looks equally bogus to begin with, going back all the way to [95c55a1](https://github.com/rpm-software-management/rpm/commit/95c55a1af9fa4be56737eefa0e709ce5abe47deb). Looks like well-intended but misguided attempt to size_t type-pedantry, I'm kinda suprised this isn't my fault because I certainly did my share of those in the early days 

I believe this code is correct, but in the interest of accelerating code review, I will leave it out and include it in a separate PR.

> Also the condition aligns on the same indentation level with the following code making it unnecessary hard to read.

How would you prefer that this code be indented?

Reply:
It's the size_t in both here and pgpGrab() that looks totally bogus (the length of data in the message does NOT depend on the architecture of the computer its being executed on). Checking for the size of an int (or "4") here is probably right. 

There's no hard rule for indenting such cases except that the condition and the following code *must* be visibly separated from each other. When in doubt, `indent -kr` generally produces results close enuf for government work... And again, the kernel coding style provides excellent guidance on these matters.

Reply:
What if I switched to `uint32_t` here?  That would be a separate PR, of course.